### PR TITLE
TabController should dispatch creation in constructor. [prod-leak-fix]

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/flutter/lib/src/material/tab_controller.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import 'dart:math' as math;
 
 import 'package:flutter/widgets.dart';
 
+import '../foundation/memory_allocations.dart';
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tab_controller.dart:9]:
```suggestion
import 'package:flutter/foundation.dart';
```

**Review by chunhtai — APPROVED** [review]
LGTM, one library import suggestion

File: packages/flutter/lib/src/material/tab_controller.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/lib/src/material/tab_controller.dart:9]: Hm, I thought we prefer relative import: https://dart.dev/tools/linter-rules/prefer_relative_imports
Absolute import creates circular reference here:  memory_allocations.dart > foundation.dart -> change_notifier.dart -> memory_allocations.dart, that makes it harder to detect layering violations.

Comment (polina-c) [reply packages/flutter/lib/src/material/tab_controller.dart:9]: But, I see, there is no consistency in Flutter Framework here. I see both relative and absolute imports.

Comment (polina-c) [reply packages/flutter/lib/src/material/tab_controller.dart:9]: I personally prefer relative, so I will leave it as is, but may be we need a team level agreement and lint here :)

Comment (chunhtai) [reply packages/flutter/lib/src/material/tab_controller.dart:9]: For cross library import we use absolute imports. Only in library import we use relative import. We need to change this back, can you create a new pr for this?

Comment (polina-c) [reply packages/flutter/lib/src/material/tab_controller.dart:9]: https://github.com/flutter/flutter/pull/135337

**Pull request merged**